lib/ddtrace/contrib/semantic_logger/integration.rb

```diff
@@ -0,0 +1,48 @@
+require 'ddtrace/contrib/integration'
+require 'ddtrace/contrib/semantic_logger/configuration/settings'
+require 'ddtrace/contrib/semantic_logger/patcher'
+
+module Datadog
+  module Contrib
+    module SemanticLogger
+      # Description of SemanticLogger integration
+      class Integration
+        include Contrib::Integration
+
+        # v4 had a migration to `named_tags` instead of `payload`
+        # and has been out for almost 5 years at this point
+        # it's probably reasonable to nudge users to using modern ruby libs
+        MINIMUM_VERSION = Gem::Version.new('4.0.0')
+
+        register_as :semantic_logger
+
+        def self.version
+          Gem.loaded_specs['semantic_logger'] && Gem.loaded_specs['semantic_logger'].version
+        end
+
+        def self.loaded?
+          !defined?(::SemanticLogger::Logger).nil?
+        end
+
+        def self.compatible?
+          super && version >= MINIMUM_VERSION
+        end
+
+        # We should probably just never auto enabled log injection as part of auto instrumentation
+        # TODO: abstract out the log injection related instrumentation into it's own module so we dont
+        # keep having to do these funky workarounds
```

Comment:
I don't think we have such cases today in Ruby, but there might be a well-behaved logging library that we can always safely patch in order to inject logs correlation safely.

I understand that this is not the case here, but I don't think `We should probably just never auto enabled log injection as part of auto instrumentation` in general.